File: app/src/main/java/com/example/androidbasic/ui/activity/OnDestroyActivity.java
package com.example.androidbasic.ui.activity;

import android.os.Bundle;
import android.support.annotation.Nullable;

/**
 * Created by vikas on 27/9/16.
 */

public class OnDestroyActivity extends BaseActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        finish();
        printLog("onStart");
    }

    @Override
    protected void onStart() {
        super.onStart();
        printLog("onStart");
    }

    @Override
    protected void onResume() {
        super.onResume();
        printLog("onResume");
    }

    @Override
    protected void onPause() {
        super.onPause();
        printLog("onPause");
    }

    @Override
    protected void onStop() {
        super.onStop();
        printLog("onStop");
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        printLog("onDestroy");
    }

    @Override
    protected void onRestart() {
        super.onRestart();
        printLog("onRestart");
    }

    @Override
    public void onBackPressed() {
        super.onBackPressed();
        printLog("onBackPressed");
    }

}

File: app/src/main/java/in/vikasrathod/androidbasic/ui/activity/OnDestroyActivity.kt
package `in`.vikasrathod.androidbasic.ui.activity

import android.os.Bundle

/**
 * Created by vikas on 27/9/16.
 */

class OnDestroyActivity : BaseActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setLogTag("__")
        finish()
        printLog("onCreate")
    }

    override fun onStart() {
        super.onStart()
        printLog("onStart")
    }

    override fun onResume() {
        super.onResume()
        printLog("onResume")
    }

    override fun onPause() {
        super.onPause()
        printLog("onPause")
    }

    override fun onStop() {
        super.onStop()
        printLog("onStop")
    }

    override fun onDestroy() {
        super.onDestroy()
        printLog("onDestroy")
    }

    override fun onRestart() {
        super.onRestart()
        printLog("onRestart")
    }

    override fun onBackPressed() {
        super.onBackPressed()
        printLog("onBackPressed")
    }

}
